Mathlib/RingTheory/Ideal/Basic.lean

```diff
@@ -641,6 +641,36 @@ theorem span_pow_eq_top (s : Set α) (hs : span s = ⊤) (n : ℕ) :
   exact ⟨f x ^ (n + 1), mul_comm _ _⟩
 #align ideal.span_pow_eq_top Ideal.span_pow_eq_top
 
+lemma isPrime_of_maximally_disjoint (I : Ideal α)
+    (S : Submonoid α) (disjoint : (I : Set α) ⊓ S = ∅)
```

Comment:
What do you think of
```suggestion
    (S : Submonoid α) (disjoint : Disjoint (I : Set α) S)
```
and similarly for `maximally_disjoint`?

Reply:
I think `Disjoint` isn't very nice definitionally (requires [disjoint_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Disjoint.html#disjoint_iff) to convert to the current form), and `¬ Disjoint` would be a bit awkward (`((I : Set α) ∩ S).Nonempty` is probably definitionally most convenient), so I'm happy with the current statement. Note that `∩` should be used instead of `⊓` since these are sets.

Reply:
The point of `Disjoint` is precisely to normalise alll those context-specific uses to a single canonical spelling, though.
You will remark that `(I : Set α) ⊓ S = ∅` is actually very bad definitionally. You can't apply it nor have dot notation on it!

I am strongly in favour of Eric's suggestion.

Reply:
I agree, we should consistently use `Disjoint` here. See also this analogous Zulip discussion on `IsCoprime`: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/coprime.20ideal/near/395055642.